hub.py

```diff
@@ -388,11 +389,18 @@ async def deploy(self, auth_provider, proxy_secret_key, test_notebook_path=None)
             print(f"Running {' '.join(cmd)}")
             subprocess.check_call(cmd)
 
-            if test_notebook_path:
-                try_idx = 1
+            if not skip_hub_health_test:
+
                 if self.spec['template'] != 'base-hub':
                     service_api_token = generated_values["base-hub"]["jupyterhub"]["hub"]["services"]["hub-health"]["apiToken"]
                 else:
                     service_api_token = generated_values["jupyterhub"]["hub"]["services"]["hub-health"]["apiToken"]
-                print("Starting hub health validation...")
-                await self.check_hub_health(test_notebook_path, service_api_token)
+
+                exit_code = pytest.main([
+                    "-s", "-v", "tests",
+                    "--hub-name", self.spec["name"],
+                    "--cluster-name", self.cluster.spec["name"],
+                    "--api-token", service_api_token
+                ])
+                if exit_code:
+                    raise(RuntimeError)
```

Comment:
Does pytest print output to stdout? That way we can see which tests failed and debug.

Reply:
Yes it does. It outputs everything we print inside the tests and the exception the test fails with.

Sample output for a passed test:
![test-passed](https://user-images.githubusercontent.com/7579677/104571199-7a5f2300-565b-11eb-95fa-fadff09eff2b.png)

Sample output for a failed test:
![failed-test](https://user-images.githubusercontent.com/7579677/104572579-87304680-565c-11eb-94a7-4c15cc18ea1e.png)